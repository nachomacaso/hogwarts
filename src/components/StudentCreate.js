import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import MenuItem from '@material-ui/core/MenuItem';
import TextField from '@material-ui/core/TextField';
import Switch from '@material-ui/core/Switch';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import Divider from '@material-ui/core/Divider';
import Grid from '@material-ui/core/Grid';
import Checkbox from '@material-ui/core/Checkbox';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Typography from '@material-ui/core/Typography';
import '../styles/index.css';

const styles = theme => ({
  container: {
    display: 'flex',
    flexWrap: 'wrap',
    padding: '20px',
  },
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
    width: 200,
  },
  dense: {
    marginTop: 19,
  },
  menu: {
    width: 200,
  },
  title: {
    background: '#7280ce',
    color: '#FFF',
    height: '40px',
    paddingLeft: '40px',
    paddingTop: '8px',
  },
  paper: {
    width: '800px',
    display: 'block',
    marginLeft: 'auto',
    marginRight: 'auto',
  },
  button: {
    float: 'right',
  },
  subHeader: {
    width: '100%',
  },
  grid: {
    textAlign: 'left',
  }
});

const houses = [
  {
    value: 'Gryffindor',
    label: 'Gryffindor Tower'
  },
  {
    value: 'Hufflepuff',
    label: 'Hufflepuff Basement'
  },
  {
    value: 'Ravenclaw',
    label: 'Ravenclaw Tower'
  },
  {
    value: 'Slytherin',
    label: 'Slytherin Dungeon'
  },
];

const courseNames = [ 
  "Transfiguration", 
  "Charms", 
  "Potions", 
  "History of Magic", 
  "Defense Against the Dark Arts", 
  "Herbology" 
];

class StudentCreate extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      age: '',
      house: 'Gryffindor',
      muggle: 'false',
      courses: [
        { id: "1", name: "Transfiguration", enrolled: 'false' },
        { id: "2", name: "Charms", enrolled: 'false' },
        { id: "3", name: "Potions", enrolled: 'false'},
        { id: "4", name: "History of Magic", enrolled: 'false' },
        { id: "5", name: "Defense Against the Dark Arts", enrolled: 'false' },
        { id: "6", name: "Herbology", enrolled: 'false' }
      ] 
    };
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  

  handleChange = name => event => {
    if(name === 'muggle') {
      this.setState({ [name]: event.target.checked.toString() });
    }
    else if(courseNames.indexOf(name) > -1) {
      let stateCopy = Object.assign({}, this.state);
      let copyCourses = stateCopy.courses;
      let updatedCourse  = copyCourses.find(x => x.name === name);
      updatedCourse.enrolled = event.target.checked.toString();
      let idx = copyCourses.findIndex(obj => obj.id === updatedCourse.id);
      copyCourses[idx] = updatedCourse;
      this.setState({ courses: copyCourses });
    }
    else {
      this.setState({ [name]: event.target.value });
    }
  }

  handleSubmit() {
    let newStudent = this.state;
    let studentData = localStorage.getItem('students');
    let copyStudents = JSON.parse(studentData);
    let idx = copyStudents[copyStudents.length - 1].id;
    newStudent['id'] = (parseInt(idx) + 1).toString();
    copyStudents.push(newStudent);
    localStorage.setItem('students', JSON.stringify(copyStudents));
  }

  render() {
    const { classes } = this.props;
    const { name, age, house, muggle, courses } = this.state;

    return (
      <div>
        <div>
          <Typography className={classes.title} color="primary" align="left" variant="subtitle1">
            Create A New Student
          </Typography>
        </div>

        <Paper className={classes.paper} elevation={1}>
          <form className={classes.container} noValidate autoComplete="off">
            <div style={{paddingBottom: '20px'}}>
              <Typography variant="h6" color="primary" align="left" style={{width: '100%'}} gutterBottom>
                Enrollment Info
              </Typography>
        
              <TextField
                id="name"
                label="Full Name"
                className={classes.textField}
                value={name}
                onChange={this.handleChange('name')}
                margin="normal"
                variant="outlined"
              />

              <TextField
                id="age"
                label="Age"
                className={classes.textField}
                value={age}
                onChange={this.handleChange('age')}
                margin="normal"
                variant="outlined"
                style={{width: '150px'}}
              />

              <TextField
                id="house"
                select
                label="Select a House"
                className={classes.textField}
                value={house}
                onChange={this.handleChange('house')}
                SelectProps={{
                  MenuProps: {
                    className: classes.menu,
                  },
                }}
                margin="normal"
                variant="outlined"
              >
                {houses.map(h => (
                  <MenuItem key={h.value} value={h.value}>
                    {h.label}
                  </MenuItem>
                ))}
              </TextField>

              <FormControlLabel
                control={
                  <Switch
                    checked={ muggle === 'true' ? true : false }
                    onChange={this.handleChange('muggle')}
                    value={muggle}
                  />
                }
                label="Muggle-Born?"
                labelPlacement="top"
              />

            </div>
            
            <Divider className={classes.divider} />
            <Typography variant="h6" align="left" color="primary" style={{width: '100%', marginTop: '20px'}} gutterBottom>
              Choose Classes
            </Typography>

            <Grid container spacing={32}>
              {courses.map(course => (
                <Grid item xs={4} className={classes.grid} key={course.id}>
                  <FormControlLabel
                    control={
                      <Checkbox onChange={this.handleChange(course.name)} 
                                value={course.enrolled}
                      />
                    }
                    label={course.name}
                  />
                </Grid>   
              ))}
            </Grid>
            <div style={{width: '100%', padding: '15px'}}>
              <Button variant="contained" color="secondary" onClick={this.handleSubmit} className={classes.button} id="create-submit" href='/students'>
                Create
              </Button>
            </div>

          </form>
          
        </Paper>
      </div>
    );
  }
}

StudentCreate.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(StudentCreate);