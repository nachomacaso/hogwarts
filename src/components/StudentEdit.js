import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import MenuItem from '@material-ui/core/MenuItem';
import TextField from '@material-ui/core/TextField';
import Switch from '@material-ui/core/Switch';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import Divider from '@material-ui/core/Divider';
import Grid from '@material-ui/core/Grid';
import Checkbox from '@material-ui/core/Checkbox';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Typography from '@material-ui/core/Typography';

const styles = theme => ({
  container: {
    display: 'flex',
    flexWrap: 'wrap',
    padding: '20px',
  },
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
    width: 200,
  },
  dense: {
    marginTop: 19,
  },
  menu: {
    width: 200,
  },
  title: {
    background: '#7280ce',
    color: '#FFF',
    height: '40px',
    paddingLeft: '40px',
    paddingTop: '8px',
  },
  paper: {
    width: '800px',
    display: 'block',
    marginLeft: 'auto',
    marginRight: 'auto',
  },
  button: {
    float: 'right',
  },
  subHeader: {
    width: '100%',
  },
  grid: {
    textAlign: 'left',
  }
});

const houses = [
  {
    value: 'Gryffindor',
    label: 'Gryffindor Tower'
  },
  {
    value: 'Hufflepuff',
    label: 'Hufflepuff Basement'
  },
  {
    value: 'Ravenclaw',
    label: 'Ravenclaw Tower'
  },
  {
    value: 'Slytherin',
    label: 'Slytherin Dungeon'
  },
];

const courseNames = [ 
  "Transfiguration", 
  "Charms", 
  "Potions", 
  "History of Magic", 
  "Defense Against the Dark Arts", 
  "Herbology" 
];

class StudentEdit extends React.Component {
  componentWillMount() {
    let id = this.props.match.params.id
    let studentData = localStorage.getItem('students');
    let copyStudents = JSON.parse(studentData);
    let student = copyStudents.find(x => x.id === id);
    this.setState(student);
  }

  handleChange = name => event => {
    if(name === 'muggle') {
      this.setState({ [name]: event.target.checked.toString() });
    }
    else if(courseNames.indexOf(name) > -1) {
      let stateCopy = Object.assign({}, this.state);
      let copyCourses = stateCopy.courses;
      let updatedCourse  = copyCourses.find(x => x.name === name);
      updatedCourse.enrolled = event.target.checked.toString();
      let idx = copyCourses.findIndex(obj => obj.id === updatedCourse.id);
      copyCourses[idx] = updatedCourse;
      this.setState({ courses: copyCourses });
    }
    else {
      this.setState({ [name]: event.target.value });
    }
  };

  handleSubmit = () => {
    const { id, name, age, house, muggle, courses } = this.state;
    let studentData = localStorage.getItem('students');
    let copyStudents = JSON.parse(studentData);
    let idx = copyStudents.findIndex((obj => obj.id === id));
    
    copyStudents[idx].name = name;
    copyStudents[idx].age = age;
    copyStudents[idx].house = house;
    copyStudents[idx].muggle = muggle;
    copyStudents[idx].courses = courses;

    localStorage.setItem('students', JSON.stringify(copyStudents));
  };

  render() {
    const { classes } = this.props;
    const { name, age, house, muggle, courses } = this.state;

    return (
      <div>
        <div>
          <Typography className={classes.title} color="primary" align="left" variant="subtitle1">
            Edit Student:  {name}
          </Typography>
        </div>

        <Paper className={classes.paper} elevation={1}>
          <form className={classes.container} noValidate autoComplete="off">
            <div style={{paddingBottom: '20px'}}>
              <Typography variant="h6" color="primary" align="left" style={{width: '100%'}} gutterBottom>
                Enrollment Info
              </Typography>
        
              <TextField
                id="name"
                label="Full Name"
                className={classes.textField}
                value={name}
                onChange={this.handleChange('name')}
                margin="normal"
                variant="outlined"
              />

              <TextField
                id="age"
                label="Age"
                className={classes.textField}
                value={age}
                onChange={this.handleChange('age')}
                margin="normal"
                variant="outlined"
                style={{width: '150px'}}
              />

              <TextField
                id="house"
                select
                label="Select a House"
                className={classes.textField}
                value={house}
                onChange={this.handleChange('house')}
                SelectProps={{
                  MenuProps: {
                    className: classes.menu,
                  },
                }}
                margin="normal"
                variant="outlined"
              >
                {houses.map(h => (
                  <MenuItem key={h.value} value={h.value}>
                    {h.label}
                  </MenuItem>
                ))}
              </TextField>

              <FormControlLabel
                control={
                  <Switch
                    checked={ muggle === 'true' ? true : false }
                    onChange={this.handleChange('muggle')}
                    value={muggle}
                  />
                }
                label="Muggle-Born?"
                labelPlacement="top"
              />

            </div>
            
            <Divider className={classes.divider} />
            <Typography variant="h6" align="left" color="primary" style={{width: '100%', marginTop: '20px'}} gutterBottom>
              Choose Classes
            </Typography>

            <Grid container spacing={32}>
              {courses.map(course => (
                <Grid item xs={4} className={classes.grid} key={course.id}>
                  <FormControlLabel
                    control={
                      <Checkbox 
                        checked={course.enrolled === 'true' ? true : false}
                        onChange={this.handleChange(course.name)} 
                        value={course.enrolled}
                      />
                    }
                    label={course.name}
                  />
                </Grid>   
              ))}
            </Grid>
            <div style={{width: '100%', padding: '15px'}}>
              <Button variant="contained" color="secondary" onClick={this.handleSubmit} className={classes.button} href="/students">
                Edit
              </Button>
            </div>

          </form>
          
        </Paper>
      </div>
    );
  }
}

StudentEdit.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(StudentEdit);