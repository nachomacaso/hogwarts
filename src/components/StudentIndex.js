import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Button from '@material-ui/core/Button';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import Fab from '@material-ui/core/Fab';
import Link from '@material-ui/core/Link';
import AddIcon from '@material-ui/icons/Add';
import '../styles/index.css';

const styles = {
  root: {
    flexGrow: 1,
  },
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  title: {
    background: '#7280ce',
    color: '#FFF',
    height: '40px',
    paddingLeft: '40px',
    paddingTop: '8px',
  },
  margin : {
    marginLeft: '20px',
    marginTop: '-7px',
    height: '30px',
    width: '30px',
    minHeight: '30px',
  }
};

class StudentIndex extends Component {
  constructor(props) {
    super(props);
    this.state = { students: [] };
  }

  componentWillMount = () => {
    let defaultStudents = [
      { 
        id: '1', name: 'Harry Potter', house: 'Gryffindor', age: '11', muggle: 'false',
        courses: [
          { id: "1", name: "Transfiguration", enrolled: 'false' },
          { id: "2", name: "Charms", enrolled: 'true' },
          { id: "3", name: "Potions", enrolled: 'true' },
          { id: "4", name: "History of Magic", enrolled: 'true' },
          { id: "5", name: "Defense Against the Dark Arts", enrolled: 'false' },
          { id: "6", name: "Herbology", enrolled: 'false' }
        ] 
      },
      { 
        id: '2', name: 'Ronald Weasley', house: 'Gryffindor', age: '11', muggle: 'false',
        courses: [
          { id: "1", name: "Transfiguration", enrolled: 'true' },
          { id: "2", name: "Charms", enrolled: 'true' },
          { id: "3", name: "Potions", enrolled: 'false'},
          { id: "4", name: "History of Magic", enrolled: 'false' },
          { id: "5", name: "Defense Against the Dark Arts", enrolled: 'true' },
          { id: "6", name: "Herbology", enrolled: 'false' }
        ] 
      },
      { 
        id: '3', name: 'Neville Longbottom', house: 'Gryffindor', age: '11', muggle: 'false',
        courses: [
          { id: "1", name: "Transfiguration", enrolled: 'true' },
          { id: "2", name: "Charms", enrolled: 'true' },
          { id: "3", name: "Potions", enrolled: 'false'},
          { id: "4", name: "History of Magic", enrolled: 'true' },
          { id: "5", name: "Defense Against the Dark Arts", enrolled: 'false' },
          { id: "6", name: "Herbology", enrolled: 'false' }
        ] 
      },
      { 
        id: '4', name: 'Hermione Granger', house: 'Gryffindor', age: '11', muggle: 'true',
        courses: [
          { id: "1", name: "Transfiguration", enrolled: 'true' },
          { id: "2", name: "Charms", enrolled: 'false' },
          { id: "3", name: "Potions", enrolled: 'false'},
          { id: "4", name: "History of Magic", enrolled: 'true' },
          { id: "5", name: "Defense Against the Dark Arts", enrolled: 'false' },
          { id: "6", name: "Herbology", enrolled: 'true' }
        ] 
      },
      { 
        id: '5', name: 'Draco Malfoy', house: 'Slytherin', age: '11', muggle: 'false',
        courses: [
          { id: "1", name: "Transfiguration", enrolled: 'true' },
          { id: "2", name: "Charms", enrolled: 'false' },
          { id: "3", name: "Potions", enrolled: 'true'},
          { id: "4", name: "History of Magic", enrolled: 'false' },
          { id: "5", name: "Defense Against the Dark Arts", enrolled: 'true' },
          { id: "6", name: "Herbology", enrolled: 'false' }
        ]
      },
      { 
        id: '6', name: 'Vincent Crabbe', house: 'Slytherin', age: '11', muggle: 'false',
        courses: [
          { id: "1", name: "Transfiguration", enrolled: 'false' },
          { id: "2", name: "Charms", enrolled: 'false' },
          { id: "3", name: "Potions", enrolled: 'true'},
          { id: "4", name: "History of Magic", enrolled: 'true' },
          { id: "5", name: "Defense Against the Dark Arts", enrolled: 'true' },
          { id: "6", name: "Herbology", enrolled: 'false' }
        ] 
      },
      { 
        id: '7', name: 'Gregory Goyle', house: 'Slytherin', age: '11', muggle: 'false',
        courses: [
          { id: "1", name: "Transfiguration", enrolled: 'true' },
          { id: "2", name: "Charms", enrolled: 'true' },
          { id: "3", name: "Potions", enrolled: 'false'},
          { id: "4", name: "History of Magic", enrolled: 'true' },
          { id: "5", name: "Defense Against the Dark Arts", enrolled: 'false' },
          { id: "6", name: "Herbology", enrolled: 'false' }
        ] 
      },
      { 
        id: '8', name: 'Luna Lovegood', house: 'Ravenclaw', age: '12', muggle: 'false',
        courses: [
          { id: "1", name: "Transfiguration", enrolled: 'false' },
          { id: "2", name: "Charms", enrolled: 'false' },
          { id: "3", name: "Potions", enrolled: 'false'},
          { id: "4", name: "History of Magic", enrolled: 'true' },
          { id: "5", name: "Defense Against the Dark Arts", enrolled: 'true' },
          { id: "6", name: "Herbology", enrolled: 'true' }
        ] 
      },
      { 
        id: '9', name: 'Tom Riddle', house: 'Slytherin', age: '31', muggle: 'false',
        courses: [
          { id: "1", name: "Transfiguration", enrolled: 'false' },
          { id: "2", name: "Charms", enrolled: 'true' },
          { id: "3", name: "Potions", enrolled: 'true'},
          { id: "4", name: "History of Magic", enrolled: 'true' },
          { id: "5", name: "Defense Against the Dark Arts", enrolled: 'false' },
          { id: "6", name: "Herbology", enrolled: 'false' }
        ] 
      }
    ];

    let defaultCourses = [
      { id: '1', name: 'Transfiguration' }, 
      { id: '2', name: 'Charms' }, 
      { id: '3', name: 'Potions' }, 
      { id: '4', name: 'History of Magic' }, 
      { id: '5', name: 'Defense Against the Dark Arts' }, 
      { id: '6', name: 'Herbology' }
    ];
    
    let studentData = localStorage.getItem('students');
    let classData = localStorage.getItem('courses');
    if(!studentData || !classData) {
        localStorage.setItem('students', JSON.stringify(defaultStudents));
        studentData = localStorage.getItem('students');

        localStorage.setItem('courses', JSON.stringify(defaultCourses));
        classData = localStorage.getItem('courses');
    }

    this.setState({ students: JSON.parse(studentData) });
  }

  handleClick = (id) => {
    let path = `/students/${id}/edit`;
    this.props.history.push(path);
  }

  handleDelete = (id) => {
    let studentData = localStorage.getItem('students');
    let copyStudents = JSON.parse(studentData);
    let idx = copyStudents.findIndex((obj => obj.id === id));
    copyStudents.splice(idx, 1);
    localStorage.setItem('students', JSON.stringify(copyStudents));

    let students = localStorage.getItem('students');

    this.setState({students: JSON.parse(students) })
  }

  formatCourses = (courses) => {
    let text = '';
    courses.forEach((course) => {
      if(course.enrolled === 'true'){
        if(text.length){
          text = text + ', ' + course.name;
        }
        else {
          text += course.name
        }        
      }
    });

    return (
      <TableCell>{text}</TableCell>
    );
  }

  

  render() {
    const { classes } = this.props;
    const { students } = this.state;
    return (
      <div className="App">
        <Typography className={classes.title} color="primary" align="left" variant="subtitle1">
          All Students
          <Link href='/students/create' className={classes.link}>
            <Fab size="small" color="secondary" aria-label="Add" className={classes.margin}>
              <AddIcon />
            </Fab>
          </Link>
        </Typography>
        <Paper className={classes.root}>
          <Table className={classes.table}>
            <TableHead>
              <TableRow>
                <TableCell>Full Name</TableCell>
                <TableCell>House</TableCell>
                <TableCell>Age</TableCell>
                <TableCell>Muggle-Born?</TableCell>
                <TableCell>Courses Enrolled</TableCell>
                <TableCell>Edit Info</TableCell>
                <TableCell>Delete Student</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {students.map(student => (
                <TableRow key={student.id}>
                  <TableCell component="th" scope="row">
                    {student.name}
                  </TableCell>
                  <TableCell>{student.house}</TableCell>
                  <TableCell>{student.age}</TableCell>
                  <TableCell>{student.muggle === 'true' ? 'Yes' : 'No'}</TableCell>
                  {this.formatCourses(student.courses)}
                  <TableCell>
                    <Button variant="contained" color="primary" size="small" onClick={() => this.handleClick(student.id)} className={classes.button}>
                      Edit
                    </Button>
                  </TableCell>
                  <TableCell>
                    <Button variant="contained" color="secondary" size="small" onClick={() => this.handleDelete(student.id)} className={classes.button}>
                      Delete
                    </Button>
                  </TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </Paper>
      </div>
    );
  }
}

export default withStyles(styles)(StudentIndex);