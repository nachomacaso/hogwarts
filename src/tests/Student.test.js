import React from 'react';
import { shallow, mount } from 'enzyme';
import StudentIndex from '../components/StudentIndex';
import StudentCreate from '../components/StudentCreate';
import StudentEdit from '../components/StudentEdit';

describe('StudentIndex', () => {
  it('should render correctly', () => {
    const component = shallow(<StudentIndex />);
  
    expect(component).toMatchSnapshot();
  });
});

describe('StudentCreate', () => {
  it('should render correctly', () => {
    const component = shallow(<StudentCreate />);
  
    expect(component).toMatchSnapshot();
  });
});

describe('StudentEdit', () => {
  it('should render correctly', () => {
    const component = shallow(<StudentEdit />);
  
    expect(component).toMatchSnapshot();
  });
});
 
describe('StudentCreate', () => {
  it('should set storage on save button click', () => {
    
    // I've tried everyting to get this to work.  I've flip flopped between 
    // having my handleSubmit function an arrow function and a constructor-bound instance method.
    // Neither is working and I'm afraid this test will fail.

    const wrapper = mount(<StudentCreate />);
    const handleSubmit = jest.spyOn(wrapper.instance(), 'handleSubmit');
    wrapper.instance().forceUpdate();
    const index = 1;
    wrapper.find('#create-submit').at(index).simulate('click');
    expect(handleSubmit.mock.calls.length).toBe(1)
    expect(component).toMatchSnapshot();
  });    
});