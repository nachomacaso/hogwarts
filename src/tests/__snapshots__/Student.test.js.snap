// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentCreate should render correctly 1`] = `
<StudentCreate
  classes={
    Object {
      "button": "StudentCreate-button-12",
      "container": "StudentCreate-container-6",
      "dense": "StudentCreate-dense-8",
      "grid": "StudentCreate-grid-14",
      "menu": "StudentCreate-menu-9",
      "paper": "StudentCreate-paper-11",
      "subHeader": "StudentCreate-subHeader-13",
      "textField": "StudentCreate-textField-7",
      "title": "StudentCreate-title-10",
    }
  }
/>
`;

exports[`StudentEdit should render correctly 1`] = `
<StudentEdit
  classes={
    Object {
      "button": "StudentEdit-button-21",
      "container": "StudentEdit-container-15",
      "dense": "StudentEdit-dense-17",
      "grid": "StudentEdit-grid-23",
      "menu": "StudentEdit-menu-18",
      "paper": "StudentEdit-paper-20",
      "subHeader": "StudentEdit-subHeader-22",
      "textField": "StudentEdit-textField-16",
      "title": "StudentEdit-title-19",
    }
  }
/>
`;

exports[`StudentIndex should render correctly 1`] = `
<StudentIndex
  classes={
    Object {
      "grow": "StudentIndex-grow-2",
      "margin": "StudentIndex-margin-5",
      "menuButton": "StudentIndex-menuButton-3",
      "root": "StudentIndex-root-1",
      "title": "StudentIndex-title-4",
    }
  }
/>
`;
